internal/caller/frame.go

```diff
@@ -0,0 +1,37 @@
+package caller
+
+import (
+	"runtime"
+	"strings"
+)
+
+// Frame contains information about the caller. This is a subset of the fields
+// in runtime.Frame.
+type Frame struct {
+	Function string
+	File     string
+	Line     int
+}
+
+// GetFrame returns information about a caller function at the specified depth
+// above this function in the the call stack.
+func GetFrame(depth int) Frame {
```

Comment:
This function is only called in the test as I understand.
Is there some other use for it?
If not, should it be moved to the test or maybe test helper file?

Reply:
Thanks for pointing, in last commit I've extracted part which is common for marshalling code and for tests, and removed all irrelevant things.

Please re-review this commit.